shotgun_api3/shotgun.py

```diff
@@ -693,6 +716,37 @@ def _construct_read_parameters(self,
             params['sorts'] = sort_list
         return params
 
+    def _add_filter_presets(self, additional_filter_presets, params):
```

Comment:
I found it confusing to have so many different uses of the name "params" in this method. What do you think about not passing in params to _add_filter_presets? Then it could instead return a dict that is merged back into params at line 699?